aepp/connector.py

```diff
@@ -45,36 +58,33 @@ def __init__(
             )
         self.config = deepcopy(config_object)
         self.header = deepcopy(header)
+        self.endpoints = deepcopy(endpoints)
         self.loggingEnabled = loggingEnabled
         self.logger = logger
         self.retry = retry
         if self.config["token"] == "" or time.time() > self.config["date_limit"]:
-            if "aepScope" in kwargs.keys() and "privacyScope" in kwargs.keys():
-                token_with_expiry = self.get_token_and_expiry_for_config(
+            if self.config["private_key"] is not None or self.config["pathToKey"] is not None:
+                token_info = self.get_jwt_token_and_expiry_for_config(
```

Comment:
I cannot comment to the correct line but I found out through an error being generated here. 
In the `configs.py` file, you need to add `"private_key": self.privateKey,` on line ~264 in order to have that reference when using an instance of the `ConnectObject` class. 
The same way, you need to extend the global config with `imsEndpoint`, `jwtTokenEndpoint` and `oathTokenEndpoint`.
I missed to update the type accepted for all methods on the `connector` module, to accept either of them.  
Normally you never used it because you pass through the init of a class and that init takes care of changing the ConnectObject class to a dict, but we need to keep the keys used in config in sync.

Reply:
@cmenguy : I created a new branch  https://github.com/pitchmuc/aepp/tree/non-user-token where I merged your modification and fixed the issues I discussed above.
I cannot really change your PR to fix it as far as I know so you can review it, try to merge it and let me know if that still works for you. 

We may want to support instance of Service Token in the long run, so you can have them store in class instead of always having to import them whenever you need to use it. 
We would need to extend the ConnectObject class for that, but I do not think that would be more difficult now.